gcc/testsuite/rust/compile/issue-1129-2.rs

```diff
@@ -0,0 +1,22 @@
+// { dg-additional-options "-w" }
+pub trait Hasher {
+    fn finish(&self) -> u64;
+    fn write(&mut self, bytes: &[u8]);
+    fn write_u8(&mut self, i: u8) {
+        self.write(&[i])
+    }
+}
```

Comment:
Oh interesting, this does not ICE in the reachability pass since there is an implementation of the trait

Reply:
Ah maybe there is something missing for error handling in type-bound-predicates